### ISSUE TITLE
Nightly builds are failing

### ISSUE THREAD

**Issue by binarylogic** [timberio/vector#3311]
As seen [here](https://github.com/timberio/vector/runs/934492571?check_suite_focus=true):

```
build-armv7-unknown-linux-musleabihf_1            | error[E0512]: cannot transmute between types of different sizes, or dependently-sized types
build-armv7-unknown-linux-musleabihf_1            |  --> /opt/rust/cargo/registry/src/github.com-1ecc6299db9ec823/cidr-utils-0.4.2/src/cidr/v4/ipv4_cidr_error.rs:4:45
build-armv7-unknown-linux-musleabihf_1            |   |
build-armv7-unknown-linux-musleabihf_1            | 4 | #[derive(Debug, Copy, Clone, Eq, PartialEq, Ordinalize)]
build-armv7-unknown-linux-musleabihf_1            |   |                                             ^^^^^^^^^^
build-armv7-unknown-linux-musleabihf_1            |   |
build-armv7-unknown-linux-musleabihf_1            |   = note: source type: `std::mem::Discriminant<cidr::v4::ipv4_cidr_error::Ipv4CidrError>` (32 bits)
build-armv7-unknown-linux-musleabihf_1            |   = note: target type: `u64` (64 bits)
build-armv7-unknown-linux-musleabihf_1            |   = note: this error originates in a derive macro (in Nightly builds, run with -Z macro-backtrace for more info)
build-armv7-unknown-linux-musleabihf_1            | 
build-armv7-unknown-linux-musleabihf_1            | error[E0512]: cannot transmute between types of different sizes, or dependently-sized types
build-armv7-unknown-linux-musleabihf_1            |  --> /opt/rust/cargo/registry/src/github.com-1ecc6299db9ec823/cidr-utils-0.4.2/src/cidr/v6/ipv6_cidr_error.rs:4:45
build-armv7-unknown-linux-musleabihf_1            |   |
build-armv7-unknown-linux-musleabihf_1            | 4 | #[derive(Debug, Copy, Clone, Eq, PartialEq, Ordinalize)]
build-armv7-unknown-linux-musleabihf_1            |   |                                             ^^^^^^^^^^
build-armv7-unknown-linux-musleabihf_1            |   |
build-armv7-unknown-linux-musleabihf_1            |   = note: source type: `std::mem::Discriminant<cidr::v6::ipv6_cidr_error::Ipv6CidrError>` (32 bits)
build-armv7-unknown-linux-musleabihf_1            |   = note: target type: `u64` (64 bits)
build-armv7-unknown-linux-musleabihf_1            |   = note: this error originates in a derive macro (in Nightly builds, run with -Z macro-backtrace for more info)
build-armv7-unknown-linux-musleabihf_1            | 
build-armv7-unknown-linux-musleabihf_1            | error[E0512]: cannot transmute between types of different sizes, or dependently-sized types
build-armv7-unknown-linux-musleabihf_1            |  --> /opt/rust/cargo/registry/src/github.com-1ecc6299db9ec823/cidr-utils-0.4.2/src/cidr/ip_cidr_error.rs:6:45
build-armv7-unknown-linux-musleabihf_1            |   |
build-armv7-unknown-linux-musleabihf_1            | 6 | #[derive(Debug, Copy, Clone, Eq, PartialEq, Ordinalize)]
build-armv7-unknown-linux-musleabihf_1            |   |                                             ^^^^^^^^^^
build-armv7-unknown-linux-musleabihf_1            |   |
build-armv7-unknown-linux-musleabihf_1            |   = note: source type: `std::mem::Discriminant<cidr::ip_cidr_error::IpCidrError>` (32 bits)
build-armv7-unknown-linux-musleabihf_1            |   = note: target type: `u64` (64 bits)
build-armv7-unknown-linux-musleabihf_1            |   = note: this error originates in a derive macro (in Nightly builds, run with -Z macro-backtrace for more info)
build-armv7-unknown-linux-musleabihf_1            | 
build-armv7-unknown-linux-musleabihf_1            | error: aborting due to 3 previous errors
build-armv7-unknown-linux-musleabihf_1            | 
build-armv7-unknown-linux-musleabihf_1            | For more information about this error, try `rustc --explain E0512`.
build-armv7-unknown-linux-musleabihf_1            | error: could not compile `cidr-utils`.
build-armv7-unknown-linux-musleabihf_1            | 
build-armv7-unknown-linux-musleabihf_1            | To learn more, run the command again with --verbose.
build-armv7-unknown-linux-musleabihf_1            | warning: build failed, waiting for other jobs to finish...
build-armv7-unknown-linux-musleabihf_1            | error: build failed
make: *** [build-armv7-unknown-linux-musleabihf] Error 101
```

**Comment by MOZGIII** [comment]
Nightlies are still broken:
https://github.com/timberio/vector/actions?query=workflow%3Anightly

**Comment by jszwedko** [comment]
Fixed by #3437 #3432 #3415
